### ISSUE TITLE
Add support for querying dark/light backgrounds

### ISSUE THREAD

**Issue by dagwieers** [dagwieers/dstat#99]
Just learned this is possible. Dstat should try to do the right thing by default: 
https://lwn.net/Articles/665053/

**Comment by scottchiefbaker** [comment]
```
echo -e "\033]11;?\007"
```

On the three terminals I've tested returns `\e]11;rgb:0000/0000/0000\a`, which should be relatively easy to parse. However, when I try and capture it in a script I get `"\e]11;?\a\n"` which translates to "I don't know what color the bg is". This may have to do with it not being an actual terminal?

Can you capture this info in Python? This would be a good candidate for that `--black-on-white`.

**Comment by dagwieers** [comment]
I noticed that the Travis output shows a "7" at the start of the line.
This is also related to the Travis terminal not supporting the complete set.
Not sure what we can test though. It's not always correct either.

E.g. Synology terminal emulation reports not supporting colors, but it works fine with `--color`.
So it's nice to have the switches just-in-case.

Improvements to auto-detection is welcomed, as long as people can still voice their preference.

**Comment by petef4** [comment]
Various techniques for discovery are captured at https://github.com/rocky/bash-term-background

Both terminator (my preferred terminal) and gnome-terminal (the system default) return "Can't decide". The detection did work with xterm.

It may be possible to have a better guess at gnome-terminal. This is a start.

    $ grep -A1 background_color ~/.gconf/apps/gnome-terminal/profiles/Default/%gconf.xml
        <entry name="background_color" mtime="1467622603" type="string">
            <stringvalue>#FFFFFFFFDDDD</stringvalue>

Similarly for terminator, which I have customized to Black on white. Before I did that there was no entry for background_color.

    $ grep background_color ~/.config/terminator/config 
        background_color = "#ffffff"

I am guessing that there are system wide defaults for these plus whatever is canned into the programs. Then there are themes to consider.
